test/views/vatDetails/VatDetailsViewSpec.scala

```diff
@@ -106,7 +116,16 @@ class VatDetailsViewSpec extends ViewBaseSpec {
     currentYear,
     paymentError = true,
     returnObligationError = true,
-    isNonMTDfBUser = None
+    isNonMTDfBUser = None,
+    customerInfoError = true
+  )
+  val customerNonMtdModel = VatDetailsViewModel(
```

Comment:
Instead of making a new model here could you use the `copy` method. For example if you use `detailsModel.copy()` and then just provide the one parameter that you are changing (here it would be `isNonMTDfBUser = Some(true)`)

Ideally the other models here would make use of copy as it's getting a bit bloated but don't worry about those, that could be a future refactor.

Reply:
Still unresolved